quickwit/quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -376,8 +376,9 @@ impl MergeExecutor {
             "Delete tasks can be applied only on one split."
         );
         assert_eq!(tantivy_dirs.len(), 1);
-        let split = splits.pop().unwrap();
+        let split = &splits[0];
         let index_metadata = self.metastore.index_metadata(&split.index_id).await?;
+        ctx.record_progress();
```

Comment:
It seems you want to protect futures here instead. `protect_future` should call `record_progress` after the future returns.

Reply:
oh nice, I did not know that.

Reply:
I'm not saying it does call `record_progress` but it probably should.

Reply:
Ok. For a simple call like `index_metadata`, I'm not fond of using `protect_future` here, we generally reserve it for bigger requests.

The issue is always the same with the actor timeout... we are in the dark and it's not always easy to guess which piece of code is responsible for the timeout.

Reply:
if you think you don't need `protect_future` then you don't need `record_progress` :D

Reply:
I don't know... I need to rerun some deletes. Let's remove them for now and see what's happening.

Reply:
I've got 2 calls to the metastore, that's probably too much. Having one record progress between the two seems fair.

Reply:
I removed them. Let's wait and see. At least now these actors are supervised.